test/Microsoft.ML.Tests/Scenarios/SentimentPredictionTests.cs

```diff
@@ -163,38 +175,71 @@ public void CrossValidateSentimentModelTest()
             Assert.True(predictions.ElementAt(1).Sentiment.IsTrue);
         }
 
-        private void ValidateBinaryMetrics(BinaryClassificationMetrics metrics)
+        private void ValidateBinaryMetrics(BinaryClassificationMetrics metrics, bool useLightGBM = false)
```

Comment:
>, bool useLightGBM = false [](start = 78, length = 26)

This is really weird design.
Why can't you just have two different methods?

Reply:
This test will go away after a month when the old API is removed. 

---
In reply to: [199234671](https://github.com/dotnet/machinelearning/pull/452#discussion_r199234671) [](ancestors = 199234671)

Reply:
Will it go away, or be replaced with something calling the new API?

Are we saying we don't care about quality and comprehensibility of the tests over the existing API, because the plan is to do away with the existing API? Would that be a sensible plan given that we are still on the hook to support the old API?

Reply:
The test will have to be rewritten. There is no concept of Pipeline in the new API. I have just added an option to switch between LightGBM and FastTree trainer in a function that will need to be deleted.